src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -599,26 +577,28 @@ bool ColorControlServer::shouldExecuteIfOff(EndpointId endpoint, uint8_t optionM
  */
 void ColorControlServer::handleModeSwitch(EndpointId endpoint, uint8_t newColorMode)
 {
-    uint8_t oldColorMode = 0;
+    auto oldColorMode = ColorMode::kCurrentHueAndCurrentSaturation;
     Attributes::ColorMode::Get(endpoint, &oldColorMode);
 
     uint8_t colorModeTransition;
 
-    if (oldColorMode == newColorMode)
+    ColorControl::ColorMode newColorModeEnum                = static_cast<ColorControl::ColorMode>(newColorMode);
```

Comment:
Again, this cast is not valid if the input is actually an `EnhancedColorModeEnum` value.